### PR TITLE
subscription: Fix issue in bundle transfer

### PR DESCRIPTION
**Pull request by sbrossie**
In cases where we have a BCD_UPADTE event, the code would throw an `java.lang.IndexOutOfBoundsException` because the event was not correctly handled.
This fix makes sure that when we have both a CREATE and BCD_UPADTE prior the transfer date, the code does not throw exception and also transfers the BCD alignment.

### CODE REVIEW

File: subscription/src/main/java/org/killbill/billing/subscription/api/transfer/DefaultSubscriptionBaseTransferApi.java [hunk 1]
```diff
@@ -114,66 +116,76 @@ private SubscriptionBaseEvent createEvent(final boolean firstEvent, final Existi
...
                 }
 
                 // Add previous event the first time if needed
-                if (prevEvent != null) {
-                    event = createEvent(firstEvent, prevEvent, subscription, transferDate, context);
-                    if (event != null) {
-                        result.add(event);
-                        firstEvent = false;
-                    }
+                if (insertEventToResult(firstEvent, prevEvent, subscription, transferDate, result)) {
```
Comment (pierre) [inline subscription/src/main/java/org/killbill/billing/subscription/api/transfer/DefaultSubscriptionBaseTransferApi.java:150]: I think I mostly managed to convince myself it works but I find the way it's written a tad off / hard to follow 😅 

We're essentially in a `for` loop and the bulk of the work is the last `insertEventToResult` (with the `cur`). The statements before are only there to insert the first `CREATE`/`BCD_CHANGE` events. Since the `if` statements are disjoints, it looks almost as if they could be inserted at any time, even though I believe they need to be insert together as soon as `cur` is after the transfer date.

File: subscription/src/test/java/org/killbill/billing/subscription/api/transfer/TestTransfer.java [hunk 2]
```diff
@@ -515,4 +517,36 @@ public void testTransferWithUncancel() throws Exception {
...
+                                                                                                         ObjectType.ACCOUNT,
+                                                                                                         this.internalCallContext.getUpdatedBy(),
+                                                                                                         this.internalCallContext.getCallOrigin(),
+                                                                                                         this.internalCallContext.getContextUserType(),
+                                                                                                         this.internalCallContext.getUserToken(),
+                                                                                                         this.internalCallContext.getTenantRecordId());
+
+        subscriptionInternalApi.updateBCD(baseSubscription.getId(), 8, clock.getUTCToday(), originalcontext);
+
+        // 2012-5-8
```
Comment (pierre) [inline subscription/src/test/java/org/killbill/billing/subscription/api/transfer/TestTransfer.java:540]: Could you add a second `BCD_CHANGE` event to make sure we pick the right one?

**Review by pierre — APPROVED** [review]

**Comment by sbrossie** [comment]
Merging as-is, will do a follow up PR with comments.
